include/pgduckdb/pgduckdb_table_am.hpp

```diff
@@ -1,7 +1,4 @@
-extern "C" {
-#include "postgres.h"
-#include "access/tableam.h"
-}
+#include "pgduckdb/pg/declarations.hpp"
```

Comment:
This was needed because we cannot define the `XactEvent` and `SubXactEvent` twice, and `tableam.h` was including `xact.h` from Postgres. Which would then cause compilation errors if `pgduckdb_table_am.hpp` was included in the same file as `pg/transactions.hpp`